### PR TITLE
resolve issue #42

### LINKED ISSUE #42: A problem with decoding vshard storage answers
**Problem 1**

`RouterMapCallRWImpl` lacks checks for len(respData) before addressing its elements.

For example, here
https://github.com/KaymeKaydex/go-vshard-router/blob/84c49a6fdfd472916879a9e6cef22c60df380139/api.go#L295

and here
https://github.com/KaymeKaydex/go-vshard-router/blob/84c49a6fdfd472916879a9e6cef22c60df380139/api.go#L298

and so on...

**Problem 2**

In `ReplicaCall`, we consider an error if len(respData) != 2. But, if fnc is completed successfully but does not return anything (i.e. returns nil), vshard does not send the second element (i.e. does not send nil):

https://github.com/KaymeKaydex/go-vshard-router/blob/84c49a6fdfd472916879a9e6cef22c60df380139/replicaset.go#L109

The fix is expected to be similar to PR #35.

### PR DESCRIPTION
**Pull request by nurzhan-saktaganov**
* ReplicaCall
	- fix decoding response (#42)
	- fix ignoring timeout while waiting for future.Get()
* Introduce Replicaset.CallAsync and struct Future
* Add tests for methods of Replicaset and Future

resolve #42

### CODE REVIEW

File: replicaset.go [hunk 1]
```diff
@@ -141,3 +135,40 @@ func (rs *Replicaset) ReplicaCall(
 		}, nil
 	}
 }
+
+type Future struct {
```
Comment (KaymeKaydex) [inline replicaset.go:139]: now future simply actually repeats the behavior of Future inside go-tarantool, but it is not used in any way, I think it’s not worth adding this yet

File: replicaset.go [hunk 2]
```diff
@@ -141,3 +135,40 @@ func (rs *Replicaset) ReplicaCall(
...
+		// suppress linter warning: lostcancel: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak (govet)
+		//nolint:govet
+		ctx, _ = context.WithTimeout(ctx, opts.Timeout)
+	}
+
+	req := tarantool.NewCallRequest(fnc).
+		Context(ctx).
+		Args(args)
+
+	var future = Future{
```
Comment (KaymeKaydex) [inline replicaset.go:169]: lets use just go-tarantool future

**Review by KaymeKaydex — CHANGES_REQUESTED** [review]

File: replicaset.go (hunk 1 above)
Comment (nurzhan-saktaganov) [reply replicaset.go]: Done

File: replicaset.go (hunk 2 above)
Comment (nurzhan-saktaganov) [reply replicaset.go]: Done

**Review by KaymeKaydex — APPROVED** [review]

**Pull request merged**